components/scream/src/physics/shoc/shoc_assumed_pdf_impl.hpp

```diff
@@ -146,17 +146,19 @@ void Functions<S,D>::shoc_assumed_pdf(
         Skew_thl.set(tsign>sp(0.2) && tsign<=sp(0.4), (((sp(1.2)*Skew_w)/sp(0.2))*(tsign-sp(0.2))));
       }
 
-      thl2_1.set(condition,
-                 ekat::min(100,
-                           ekat::max(0, (3*tmp_val_1*(1 - a*ekat::square(tmp_val_2) - (1-a)*ekat::square(tmp_val_1))
-                                         - (Skew_thl - a*ekat::cube(tmp_val_2) - (1 - a)*ekat::cube(tmp_val_1)))
-                                         /(3*a*(tmp_val_1 - tmp_val_2))))*thlsec);
-      thl2_2.set(condition,
-                 ekat::min(100,
+      if (condition.any()) {
+        thl2_1.set(condition,
+                   ekat::min(100,
+                             ekat::max(0, (3*tmp_val_1*(1 - a*ekat::square(tmp_val_2) - (1-a)*ekat::square(tmp_val_1))
+                                           - (Skew_thl - a*ekat::cube(tmp_val_2) - (1 - a)*ekat::cube(tmp_val_1)))
+                                       /(3*a*(tmp_val_1 - tmp_val_2))))*thlsec);
```

Comment:
This expression is quite complicated. Is it guaranteed that if condition is true, then the denominator is nonzero? To me, it's hard to tell. Might be that with this mod the test passes, but I can't tell if it just "happens" to pass, or if it will "always" pass, even when we add more scream tests (e.g., fully coupled scream, with real data).

Reply:
Good question. I think if `w1_2 - w1_1 > w_thresh` then `tmp_val_1 - tmp_val_2` should never be zero.